src/tools/crossgen2/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -697,6 +697,13 @@ private uint getMethodAttribsInternal(MethodDesc method)
                 result |= CorInfoFlag.CORINFO_FLG_FINAL;
             }
 
+            // Check for SIMD intrinsics
+            if (method.OwningType.IsIntrinsic && method.OwningType is MetadataType mdType && 
```

Comment:
It would be nice to add a method like this somewhere:

```
        public static bool IsVectorOfTType(DefType type)
        {
            return type.IsIntrinsic && type.Namespace == "System.Numerics" && type.Name == "Vector`1";
        }
```

and then use it for all places that do this special casing.

VectorOfTFieldLayoutAlgorithm may be a good place for it?

Reply:
Sounds reasonable. I was not sure where to put such methods.